redbeams/src/main/java/com/sequenceiq/redbeams/converter/cloud/CredentialToCloudCredentialConverter.java

```diff
@@ -0,0 +1,34 @@
+package com.sequenceiq.redbeams.converter.cloud;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.common.json.Json;
+import com.sequenceiq.redbeams.domain.Credential;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.stereotype.Component;
+
+@Component
+public class CredentialToCloudCredentialConverter implements Converter<Credential, CloudCredential> {
+
+    private static final String CREDENTIAL_ID = "id";
+
+    public CloudCredential convert(Credential credential) {
+        if (credential == null) {
+            return null;
+        }
+        Map<String, Object> fields;
+        if (credential.getId() == null) {
+            fields = isEmpty(credential.getAttributes()) ? new HashMap<>() : new Json(credential.getAttributes()).getMap();
```

Comment:
This whole converter is a little wonky. It seems that it shouldn't matter, when setting fields, what the ID is. And then later the id field is set to null in the fields map.

Reply:
I'm slightly uncomfortable modifying this -- this is a copy/paste from FreeIPA/Cloudbreak/etc, and this goes down into the SPI layer. I'm hesitant to change this at the moment.

Reply:
OK, I'm fine with leaving it for now, especially since it's a clone from elsewhere.